www/js/survey/multilabel/MultiLabelButtonGroup.tsx

```diff
@@ -94,6 +111,29 @@ const MultilabelButtonGroup = ({ trip, buttonsInline = false }) => {
   }
 
   const tripInputDetails = labelInputDetailsForTrip(timelineLabelMap[trip._id.$oid], appConfig);
+
+  // ex) my_car -> My Car
+  const getProcessedText = (key: string): string =>
+    key
+      .split('_')
+      .map((k) => k.charAt(0).toUpperCase() + k.slice(1))
+      .join(' ');
+
+  useEffect(() => {
+    // Whenever the modal opens for Mode, retrieve updated modes from the server for synchronized data
```

Comment:
I was contemplating the choice between `consistency` and `efficiency`. If we only load custom modes on the initial load, there might be inconsistency. That's why I fetch the custom modes whenever the modal opens. However, I also had concerns about unnecessary API calls. Considering your comment, I think efficiency is a better trade-off. Therefore, I will change the logic as you recommended!

Reply:
Given that these are user-specific labels, I don't think there are any issues about consistency as long as you maintain a local cache and unify. You will only run into consistency issues if we maintain a shared set of modes. Which we may want to do eventually so that, for aggregate analysis, we may not have library, Library and liBray. That is much harder to design:
- what do we do about multiple languages, for example (e.g. church, Church and iglesia, Iglesia)
- even more complicated with non-roman scripts (e.g. temple versus கோவில் versus मंदिर)
- what about bad actors who use inappropriate language in the modes (no examples here to keep it SFW, but I think you can get the idea)

So we will stick with user specific modes for this initial step. The user can fill them in the language of their choice, and if they use inappropriate language, they are the only ones affected.

So definitely stick with performance for now. When we get to the global mode table, we can think about consistency (and will likely use eventual consistency for performance).